### PR TITLE
Feature/test tour page

### PR DESCRIPTION
**Pull request by HIMDanny**
tests for tour components

### CODE REVIEW

File: client/src/components/TourInfoDialog/TourInfoDialog.test.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+jest.mock('@mui/material/Box', () => ({ children }) => <div>{children}</div>);
+jest.mock('@mui/material/Stack', () => ({ children }) => <div>{children}</div>);
+jest.mock('@mui/material/Container', () => ({ children }) => <div>{children}</div>);
+jest.mock('@mui/material/Typography', () => ({ children }) => <p>{children}</p>);
+jest.mock('@mui/material/Button', () => ({ children }) => <button>{children}</button>);
+jest.mock('@mui/material/IconButton', () => ({ children }) => <button>{children}</button>);
+jest.mock('@mui/material/Select', () => ({ children }) => <select>{children}</select>);
+jest.mock('@mui/material/MenuItem', () => ({ children }) => <option>{children}</option>);
+
+const handleClose = jest.fn();
```
Comment (Ahmed-Fahmy-Dev) [inline client/src/components/TourInfoDialog/TourInfoDialog.test.js]: you dont need that if its a snapshot test

File: client/src/components/TourInfoDialog/TourInfoDialog.test.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  test('should TourInfoDialog with close button match snapshot', () => {
+    const { asFragment } = render(
+      <TourInfoDialog
+        included={included}
+        cost={cost}
+        dates={dates}
+        details={details}
+        closeButton
+        handleClose={handleClose}
```
Comment (Ahmed-Fahmy-Dev) [inline client/src/components/TourInfoDialog/TourInfoDialog.test.js]: you can delete handleClose as we not using the hasBeenCalled matcher in this snapshot test.

**Review by AltysS — APPROVED** [review]
Well done. Approved from my side

File: client/src/components/TourInfoDialog/TourInfoDialog.test.js (hunk 1 above)
Comment (HIMDanny) [reply client/src/components/TourInfoDialog/TourInfoDialog.test.js]: I made this for future tests, but ok, no problem to delete for now

Comment (HIMDanny) [reply client/src/components/TourInfoDialog/TourInfoDialog.test.js]: done

File: client/src/components/TourInfoDialog/TourInfoDialog.test.js (hunk 2 above)
Comment (HIMDanny) [reply client/src/components/TourInfoDialog/TourInfoDialog.test.js]: done

File: client/src/pages/TourPage/TourPage.test.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import { mockAllIsIntersecting } from 'react-intersection-observer/test-utils';
+
+jest.mock('@mui/material/Box', () => 'div');
+jest.mock('@mui/material/Stack', () => ({ children }) => <div>{children}</div>);
+jest.mock('@mui/material/Container', () => ({ children }) => <div>{children}</div>);
+jest.mock('@mui/material/Typography', () => ({ children }) => <p>{children}</p>);
+jest.mock('@mui/material/Button', () => ({ children }) => <button>{children}</button>);
+jest.mock('@mui/material/Slide', () => ({ children }) => <div>{children}</div>);
+jest.mock('@mui/material/Link', () => ({ children }) => <a>{children}</a>);
+jest.mock('../../components/TourAccordion/TourAccordion.js', () => () => <div>TourAccordion</div>);
```
Comment (Ahmed-Fahmy-Dev) [inline client/src/pages/TourPage/TourPage.test.js]:
```suggestion
jest.mock('../../components/TourAccordion/TourAccordion.js', () => ({children}) => <div>{children}</div>);
```

**Review by Ahmed-Fahmy-Dev — APPROVED** [review]
perfect.

**Pull request merged**